workflow/torch/datastream.py

```diff
@@ -171,3 +435,93 @@ def test_datastream_zip():
     assert batch[0][0] == 1 and batch[0][1] == 2 and batch[0][2] == 1
     assert batch[1][0] == 3 and batch[1][1] == 4 and batch[1][2] == 5
     assert batch[2][0] == 6 and batch[2][1] == 7 and batch[2][2] == 6
+
+
+def test_datastream_merge_zip_merge():
+    '''
+    repeating because it only sometimes recreated an error that occured
+    when using mixup/mixmatch
+    '''
+
+    def RandomDatastream():
+        return Datastream(Dataset.from_subscriptable(
+            list(range(np.random.randint(1, 10)))
+        ))
+
+    def MergedDatastream():
+        return Datastream.merge([RandomDatastream(), RandomDatastream()])
+
+    def ZippedMergedDatastream():
+        return Datastream.zip([MergedDatastream(), MergedDatastream()])
+
+    for attempt in range(10):
+        print('attempt:', attempt)
+        datastream = Datastream.merge([
+            (ZippedMergedDatastream(), 1),
+            (ZippedMergedDatastream(), 5),
+        ])
+
+        it = iter(datastream.data_loader(batch_size=16, n_batches_per_epoch=10))
+        for _ in range(10):
+            print(next(it))
```

Comment:
tests should be silent unless they fail :)

Reply:
pytest only prints if there is an error so I thought this was intended. Does that change your mind?

Reply:
a thats right, it actually traps the output and only prints everything once it fails?
in that case and given the nature of what this is trying to test it's okay :)

Reply:
just to add, I thought the prints was more of something you used when you constructed the test and then just let it be left